tests/app/main/views/test_communications.py

```diff
@@ -39,88 +52,198 @@ def test_get_page_should_only_be_accessible_to_specific_user_roles(self, role, e
         assert actual_code == expected_code, "Unexpected response {} for role {}".format(actual_code, role)
 
     def test_page_shows_empty_messages_if_no_files_uploaded(self):
-        self.s3.return_value.list.side_effect = ([], [])  # Empty lists for both communication and clarification files
-        self.data_api_client.get_framework.return_value = {"frameworks": {"status": "open"}}
+        self.s3.return_value.list.side_effect = lambda *args, **kwargs: []  # return a fresh empty list every time
```

Comment:
Is someone mutating these lists? :thinking:

Reply:
Dunno. If it doesn't now, someone could change the function under test to do so next month without thinking about it, not for a minute thinking that the mocks we use here are strangely behaved - I'm just trying to make it behave as much like the real method as possible. The mocks should try not to know too much about what the function under test does and doesn't do.